server/src/v2/modules/db.js

```diff
@@ -116,59 +116,51 @@ function initialise() {
 
 initialise();
 
-export function db_addSecret(secretObject) {
+function getStatement(preparedStatement, statementParams) {
+  //Returns data
+  let row;
   try {
-    databaseFile.prepare(INSERT_SECRET_QUERY).run(secretObject.secret_id,secretObject.secret_text, secretObject.passphrase, secretObject.expiry_date, secretObject.method);
+    row = databaseFile.prepare(preparedStatement).get(statementParams);
   }
   catch (err) {
     return { data: null, code: 500, human_code: `failure, ${err}` };
   }
-  return { data: null, code: 200, human_readable_code: "Success" };
+  return { data: row, code: 200, human_readable_code: "Success" };
 }
 
-export function db_retrieveSecret(secretID) {
-  let row;
-  try{
-    row = databaseFile.prepare(GET_SECRET_QUERY).get(secretID);
+function runStatement(preparedStatement, statementParams) {
+  //Does not return data
+  try {
+    databaseFile.prepare(preparedStatement).run(statementParams);
   }
   catch (err) {
     return { data: null, code: 500, human_code: `failure, ${err}` };
   }
-  return { data: row, code: 200, human_readable_code: "Success" };
+  return { data: null, code: 200, human_readable_code: "Success" };
 }
 
-export function db_retrievePassphrase(secretID) {
-  let row;
-  try{
-    row = databaseFile.prepare(GET_PASSPHRASE_QUERY).get(secretID);
-  }
-  catch (err) {
-    return { data: null, code: 500, human_code: `failure, ${err}` };
-  }
-  return { data: row, code: 200, human_readable_code: "Success" };
+export function addSecret(secretObject) {
+  return runStatement(INSERT_SECRET_QUERY, secretObject.secret_id, secretObject.secret_text, secretObject.passphrase, secretObject.expiry_date, secretObject.method);
 }
 
-export function db_deleteSecret(secretID) {
-  try {
-    databaseFile.prepare(DELETE_SECRET_QUERY).run(secretID);
-  }
-  catch (err) {
-    return { data: null, code: 500, human_code: `failure, ${err}` };
-  }
-  return { data: null, code: 200, human_readable_code: "Success" };
+
+export function retrieveSecret(secretID) {
+  return getStatement(GET_SECRET_QUERY, secretID);
```

Comment:
I can do that, although I'm also concerned about the computational overhead that might come with constantly re-defining them every time the function is executed. The way that they are now, they should all be loaded in when the module is, as far as I know, rather than having to be constantly re-defined, as they might be if they were contained inside the function. 

I don't mind doing that for low-use functions for which the overhead would be negligible, but I am concerned about the possible performance impact that it might have if I did that for something like the secret add/removal/fetch operations.